resources/js/features/achievements/components/UnlockableAchievementAvatar/UnlockableAchievementAvatar.tsx

```diff
@@ -0,0 +1,77 @@
+import type { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { AchievementAvatar } from '@/common/components/AchievementAvatar';
+import { GameAvatar } from '@/common/components/GameAvatar';
+import type { AvatarSize } from '@/common/models';
+import { UnlockedAtLabel } from '@/features/achievements/components/UnlockedAtLabel';
+
+interface UnlockableAchievementAvatarProps {
+  achievement: App.Platform.Data.Achievement;
+  showGame?: boolean;
+  imageSize?: AvatarSize;
+}
+
+export const UnlockableAchievementAvatar: FC<UnlockableAchievementAvatarProps> = ({
+  achievement,
+  showGame = false,
+  imageSize = 48,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="mb-2 flex items-center gap-2">
+      {achievement.unlockedHardcoreAt ? (
+        <AchievementAvatar
+          {...achievement}
+          showHardcoreUnlockBorder={true}
+          // TODO: showPointsInTitle={true}
+          showLabel={false}
+          size={imageSize}
+        />
+      ) : achievement.unlockedAt ? (
+        <AchievementAvatar
+          {...achievement}
+          showHardcoreUnlockBorder={false}
+          // TODO: showPointsInTitle={true}
+          showLabel={false}
+          size={imageSize}
+        />
+      ) : (
+        <AchievementAvatar
+          {...achievement}
+          showHardcoreUnlockBorder={false}
+          badgeUnlockedUrl={achievement.badgeLockedUrl}
+          // TODO: showPointsInTitle={true}
+          showLabel={false}
+          size={imageSize}
+        />
+      )}
+
+      <div>
+        <div className="flex items-center gap-2">
+          <AchievementAvatar {...achievement} showImage={false} />
+
+          {showGame && achievement.game ? (
+            <>
+              <span>{t('from')}</span>
+              <GameAvatar {...achievement.game} showImage={false} />
+            </>
+          ) : (
+            <></>
```

Comment:
nit: it's kinda counterintuitive, but `null` is slightly better for performance than `<></>`.

Explanation: Every tag in a React component is just syntactic sugar around pure JavaScript functions. At build-time, Vite scans over all of the template code and converts everything from:
```tsx
<p className="text-lg">Hello, world!</p>
```
to:
```tsx
React.createElement("p", { className: "text-lg" }, "Hello world!");
```

`<></>` is also a component (called a Fragment), so it will be invoked at run-time in the user's browser even though it isn't doing anything:
```tsx
React.createElement(React.Fragment, null);
```
By instead returning `null`, we tell React to just skip this part of the tree.

This feedback also applies to the fragment return in line 72.

Reply:
Done. Another solution provide by google.